oasislmf/model_preparation/lookup.py

```diff
@@ -322,15 +323,25 @@ def get_custom_lookup(
                 model_name=model_info['model_id'],
                 model_version=model_info['model_version']
             )
-
-        return klc(
-            keys_data_directory=keys_data_path,
-            supplier=model_info['supplier_id'],
-            model_name=model_info['model_id'],
-            model_version=model_info['model_version'],
-            complex_lookup_config_fp=complex_lookup_config_fp,
-            output_directory=output_directory
-        )
+        elif not user_data_dir:
+            return klc(
+                keys_data_directory=keys_data_path,
+                supplier=model_info['supplier_id'],
+                model_name=model_info['model_id'],
+                model_version=model_info['model_version'],
+                complex_lookup_config_fp=complex_lookup_config_fp,
+                output_directory=output_directory
+            )
+        else:
+            return klc(
+                keys_data_directory=keys_data_path,
+                supplier=model_info['supplier_id'],
+                model_name=model_info['model_id'],
+                model_version=model_info['model_version'],
+                complex_lookup_config_fp=complex_lookup_config_fp,
+                user_data_dir=user_data_dir,
```

Comment:
Can't you just let this be `None` and avoid the extra `elif` above (and the other `elif` above that)? I would have thought that leaving out the argument would cause it to take a default value which I would expect to be `None` inside the lookup function.